src/main/java/br/com/budgetboys/pf_finance_service/adapters/outbound/repositories/income/IncomeRepositoryImpl.java

```diff
@@ -3,40 +3,46 @@
 import br.com.budgetboys.pf_finance_service.adapters.outbound.entities.JPAIncomeEntity;
 import br.com.budgetboys.pf_finance_service.domain.income.Income;
 import br.com.budgetboys.pf_finance_service.domain.income.IncomeRepository;
+import br.com.budgetboys.pf_finance_service.domain.income.IncomeResponseDTO;
+import br.com.budgetboys.pf_finance_service.utils.mappers.IncomeMapper;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 import java.util.stream.Collectors;
 
+@Repository
 public class IncomeRepositoryImpl implements IncomeRepository {
 
     private final JPAIncomeRepository jpaIncomeRepository;
 
-    public IncomeRepositoryImpl(JPAIncomeRepository jpaIncomeRepository) {
+    @Autowired
+    private final IncomeMapper incomeMapper;
+
+    public IncomeRepositoryImpl(JPAIncomeRepository jpaIncomeRepository, IncomeMapper incomeMapper) {
         this.jpaIncomeRepository = jpaIncomeRepository;
+        this.incomeMapper = incomeMapper;
     }
 
     @Override
     public Income save(Income income) {
-        JPAIncomeEntity incomeEntity = new JPAIncomeEntity(income);
-        this.jpaIncomeRepository.save(incomeEntity);
-        return new Income(incomeEntity.getId(), incomeEntity.getAmount(), incomeEntity.getCategory());
+        JPAIncomeEntity incomeEntity = this.incomeMapper.entityToJpa(income);
+        JPAIncomeEntity savedEntity = this.jpaIncomeRepository.save(incomeEntity);
+        return this.incomeMapper.jpaToEntity(savedEntity);
     }
 
     @Override
     public Income findById(UUID id) {
         Optional<JPAIncomeEntity> incomeEntity = this.jpaIncomeRepository.findById(id);
-        return incomeEntity.map(entity -> new Income(entity.getId(), entity.getAmount(), entity.getCategory())).orElse(null);
+        return incomeEntity.map(this.incomeMapper::jpaToEntity).orElse(null);
     }
 
     @Override
-    public List<Income> findAll() {
-        return this.jpaIncomeRepository.findAll()
-                .stream()
-                .map(entity -> new Income(entity.getId(), entity.getAmount(), entity.getCategory()))
-                .collect(Collectors.toList());
+    public List<IncomeResponseDTO> findAll() {
+        return this.jpaIncomeRepository.findAll().stream().map(this.incomeMapper::jpaToResponse).collect(Collectors.toList());
+
```

Comment:
Acho que aqui esteja errado? era pra retornar o DTO mesmo?
```suggestion
    public List<Income> findAll() {
        return this.jpaIncomeRepository.findAll().stream().map(this.incomeMapper::jpaToEntity).collect(Collectors.toList());
```